doc/opae-c++/opaecxx.md

```diff
@@ -0,0 +1,184 @@
+# OPAE C++ API
+## Overview
+The OPAE C++ API enables C++ developers with the means to use FPGA resources by integrating the OPAE software stack into C++ applications. It is built on top of the OPAE C API and is split up into two components:
+1. The basic or fundemental types that wrap the data structures defined in the OPAE C API.
+2. Classes that abstract FPGA resources and functions
+
+## Goals
+The following is a list of goals and design decisions made while designing this API.
+
+### Simplicity
+Keep the API as small and lightweight as possible. Features such as system validation and orchestration are beyond the scope of this.

```

Comment:
Meaning that we don't provide an orchestrator here.. but I think we would advocate that an orchestrator could be built with the C++ layer; so, we don't want to scare anyone away who may want to do that.